code/game/machinery/doors/airlock.dm

```diff
@@ -488,18 +478,21 @@ There are 9 wires.
 	return src.isWireCut(AIRLOCK_WIRE_BACKUP_POWER1) || src.isWireCut(AIRLOCK_WIRE_BACKUP_POWER2)
 
 /obj/machinery/door/airlock/proc/loseMainPower()
-	main_power_lost_until = mainPowerCablesCut() ? -1 : world.time + SecondsToTicks(60)
+	main_power_lost_until = mainPowerCablesCut() ? -1 : SecondsToTicks(60)
+	spawn(main_power_lost_until) regainMainPower()
```

Comment:
Maybe you better to use this?
```suggestion
	addtimer(CALLBACK(src, .proc/regainMainPower), main_power_lost_until)
```

Reply:
spawn is less resource intensive than timers , and i don't need the precision of one here

Reply:
> spawn is less resource intensive than timers , and i don't need the precision of one here

Spawns isuncotrollable, better to use timers

Reply:
I don't need it to be controllable , using timers here is redundant and i don't need their functionality here at all , but whatever.